python/cudf/cudf/core/column/decimal.py

```diff
@@ -29,8 +29,11 @@ def from_arrow(cls, data: pa.Array):
 
     def to_arrow(self):
         data_buf_64 = self.base_data.to_host_array().view("int64")
-        data_buf_128 = np.zeros(len(data_buf_64) * 2, dtype="int64")
+        data_buf_128 = np.empty(len(data_buf_64) * 2, dtype="int64")
         data_buf_128[::2] = data_buf_64
+        data_buf_128[1::2] = np.piecewise(
+            data_buf_64, [data_buf_64 < 0], [-1, 0]
+        )
```

Comment:
It would be good to comment on exactly what this code is doing.

Reply:
✅